### PR TITLE
Add features matrix to the README

### LINKED ISSUE #1279: Include Features Matrix
Lots of users ask in Slack what we support (and what we plan on supporting)

We should throw a simple `FEATURES.md` in the repo.

### PR DESCRIPTION
**Pull request by Sean-Der**
List most popular features/benefits of Pion WebRTC.

Resolves #1279

Co-authored-by: ZHENK <chengzhenyang@gmail.com>

### CODE REVIEW

**Comment by Sean-Der** [comment]
Would really love peoples opinion on this? Did I miss anything important? Any points I make do you think don't matter?

**Comment by Sean-Der** [comment]
@albrow @backkem @adwpc @AeroNotix @Antonito @ashellunts @at-wat @audiolion @baiyufei @billylindeman @bshi @cgojin @cloudwebrtc @cryptix @decanus @enobufs @epes @hugoArregui @jeremija @jech @leewardbound @lherman-cs @masterada @maxhawkins @mission-liao @OrlandoCo @r-novel @rviscarra @RyanGordon @saljam @scorpionknifes @seppo0010 @sgotti @steebchen @TannerGabriel @tarrencev @tekig @thinkski @trivigy @wdouglass 

Would this have improved/detracted from your first opinion of using Pion? I have gotten a lot of feedback that people have no idea what `pion/webrtc` contains so wanted to bring this to the top.

**Comment by Sean-Der** [comment]
https://github.com/pion/webrtc/blob/eef6045bb0203f0223f809cf0187571297b6adf6/README.md is a link directly to the file to see it formatted.

File: README.md [hunk 1]
```diff
@@ -33,8 +33,56 @@ We are actively planning and would love your feedback! Please create GitHub issu
...
+#### Media
+* API with direct RTP/RTCP access
+* Opus, PCM, H264, VP8 and VP9 packetizer
+* API also allows developer to pass their own packetizer
+* IVF, Ogg, H264 and Matroska provided for easy sending and saving
+* [getUserMedia](https://github.com/pion/mediadevices) implementation (Requires Cgo)
+* Easy integration with x264, libvpx, GStreamer and ffmpeg.
+* [Simulcast](https://github.com/pion/webrtc/tree/master/examples/simulcast)
+* [SVC](https://github.com/pion/rtp/blob/master/codecs/vp9_packet.go#L138)
+* [NACK](https://github.com/pion/interceptor/issues/7)
```
Comment (hugoArregui) [inline README.md:78]: why there is a link to an issue? it means not implemented?

Comment (Sean-Der) [reply README.md:78]: Almost there! It is in a PR [here](https://github.com/pion/interceptor/pull/4) but not merged yet.

Bandwidth Estimation is in Ion and I am hoping we can just move into interceptors soon!

Forward Error Correction isn't available at all yet :(

Comment (hugoArregui) [reply README.md:78]: Assuming the answer is yes (sorry, going to bed soon :joy: ), I don't think it is really clear (even if the answer is no I think it's not clear!). Also: I think it's easy to get outdated links once we close or deprecate issues. I think maybe for TODOs we can have a link at the bottom with a big global issue pointing to the current work, or maybe even to an issue search with label next-feature or something. Just my 2c!

**Comment by trivigy** [comment]
Would have changed anything for me. WebRTC is a fairly complex tech stack and usually requires decent know how. By the time I was using it full force I already knew what is inside.

File: README.md [hunk 2]
```diff
@@ -33,8 +33,56 @@ We are actively planning and would love your feedback! Please create GitHub issu
...
+### Features
+#### PeerConnection API
+* Go implementation of [webrtc-pc](https://w3c.github.io/webrtc-pc/) and [webrtc-stats](https://www.w3.org/TR/webrtc-stats/)
+* DataChannels
+* Send/Receive audio and video
+* Renegotiation
+* Plan-B and Unified Plan
+* [SettingEngine](https://pkg.go.dev/github.com/pion/webrtc/v3#SettingEngine) for Pion specific extensions
+
+#### Pure Go
```
Comment (tarrencev) [inline README.md:45]: minor feedback: this section in between the webrtc specific functionality seems a bit out of place

**Review by tarrencev — APPROVED** [review]
much better!

**Comment by ashellunts** [comment]
Definitely good improvement of README!

**Comment by jeremija** [comment]
I like this!

My two cents: the `pion/webrtc` dependencies under the `pion/` organization sometimes miss the details on why are they exist. For example, when I was starting out exploring pion and its libraries, I noticed the READMEs of `pion/rtp`, `pion/rtcp` and `pion/srtp` all looked the same, with no additional context about what each of these libraries was about.

On the other hand, the `pion/dtls` library has a nice list with features at the top of the README.

**Comment by r-novel** [comment]
Looks good to me!

**Comment by TannerGabriel** [comment]
It seems like a great improvement!

**Comment by audiolion** [comment]
Love this change 🙌

**Comment by jech** [comment]
With all due respect, Sean, I think the "feature matrix" is overselling Pion.  It might raise people's expectations too much, and then cause them to turn away from Pion when they realise that we've been overselling.

I recommend marking Simulcast and SVC as "work in progress".  I don't know what is the status of *mediadevices*, please mark it as experimental or something.  Please be clear upfront that the user must bring their own implementations of congestion control and loss recovery.

**Review by lherman-cs — APPROVED** [review]
It looks really good!

File: README.md (hunk 2 above)
Comment (Sean-Der) [reply README.md:45]: moved to the bottom!

**Comment by backkem** [comment]
> I like this!
> 
> My two cents: the `pion/webrtc` dependencies under the `pion/` organization sometimes miss the details on what they are for. For example, when I was starting out exploring pion and its libraries, I noticed the READMEs of `pion/rtp`, `pion/rtcp` and `pion/srtp` all looked the same, with no additional context about what each of these libraries were about.
> 
> On the other hand, the `pion/dtls` library has a nice list with features at the top of the README.

We split every protocol out to encourage them to be used on their own. Some didn't get much attention while others, like DTLS, grew their own vibrant community. It just happened naturally. Maybe improving the readmes of those other libraries may help attract eyes thought.

**Review by steebchen — APPROVED** [review]
LGTM; when I started with WebRTC I've had trouble what this repository exactly is and how it relates to all the others. It's pretty clear with this PR. 👍

**Comment by tekig** [comment]
Many complex terms can scare people off or raise their expectations. They may say "A lot has been done here, we can easily use this project", but they are not.

I think that people who understand will not switch to another project, because they already have a ready-made solution that works. Not sure, but people come here who don't know much about WebRTC in the beginning. The words ICE, TURN and etc wouldn't give me anything. I wanted to see an example that works and is simple so that a quick start can be done.

I think at the beginning you need to put a quick start, only then the features. An example should implement the most popular usecase, and also offer examples for less popular usecase. These examples should work in the real world so that the user does not think they have been fooled.

Sorry a little off topic.

**Comment by Sean-Der** [comment]
@tekig what do you think of pushing features down lower? We can have `Usage` and `What is WebRTC` and then features.

I think we do have some people coming to this and evaluating how polished Pion is (where I think this list helps). I think you are right though about the majority of users don't care about feature set and just want to get going.

**Comment by thinkski** [comment]
> Would this have improved/detracted from your first opinion of using Pion? I have gotten a lot of feedback that people have no idea what `pion/webrtc` contains so wanted to bring this to the top.

Having this list of features is great, although the README is starting to get busy -- maybe better to have a 'Features' page on the wiki or pion.ly and link to it from the README.md?

Personally, I learn best by example, so the example applications/demos were the most useful to get a feel for Pion.

**Comment by tekig** [comment]
> @tekig what do you think of pushing features down lower? We can have `Usage` and `What is WebRTC` and then features.

a good idea

**Comment by wdouglass** [comment]
I'm late on this, but it's a good change!

**Pull request merged**